tests/bsim/bluetooth/mesh/src/test_beacon.c

```diff
@@ -1338,38 +1357,393 @@ static void test_rx_priv_interleave(void)
+		{.net = &pp_net1},
+	};
+
+	for (size_t i = 0; i < ARRAY_SIZE(net_ctx); i++) {
+		if (beacon.pp_hash == pp_hash_calc(net_ctx[i].net, beacon.pp_random, NULL)) {
+			if (old_idx == 0xff) {
+				/* Received first Net ID advertisment */
+				old_idx = i;
+				net_ctx[i].start = k_uptime_get();
+				net_ctx[i].recv_cnt++;
+			} else if (old_idx != i) {
+				/* Received Net ID adv for new subnet */
+
+				/* Verify last Net ID adv result */
+				ASSERT_IN_RANGE(k_uptime_get() - net_ctx[old_idx].start,
+						MAX_TIMEOUT - 1000, MAX_TIMEOUT);
+				ASSERT_IN_RANGE(net_ctx[old_idx].recv_cnt,
+						9, 10);
+				net_ctx[old_idx].recv_cnt = 0;
+				old_idx = i;
+
+				/* End test when all itterations are completed */
+				if (!((test_ctx->arg.itr--) - 1)) {
+					return true;
+				}
+
+				net_ctx[i].start = k_uptime_get();
+				net_ctx[i].recv_cnt++;
+			} else {
+				/* Received another Net ID adv from same subnet*/
+				net_ctx[i].recv_cnt++;
+			}
+
+			break;
+		}
+	}
+	return false;
+}
+
+static void priv_scan_cb(const bt_addr_le_t *addr, int8_t rssi, uint8_t adv_type,
+				struct net_buf_simple *buf)
+{
+	struct priv_test_ctx *ctx = (struct priv_test_ctx *)beacon.user_ctx;
+
+	if (!pp_type_check(ctx->beacon_type, adv_type, buf)) {
+		/* Wrong message type */
+		return;
+	}
+
+	bt_addr_le_copy(&beacon.adv_addr, addr);
+
+	beacon.pp_hash = net_buf_simple_pull_le64(buf);
+	beacon.pp_random = net_buf_simple_pull_le64(buf);
+
+	if (!beacon.process_cb || beacon.process_cb(NULL, beacon.user_ctx)) {
+		k_sem_give(&observer_sem);
+	}
+}
+
+static void rx_priv_common_init(uint16_t wait)
+{
+	int err;
+
+	bt_mesh_test_cfg_set(&rx_cfg, wait);
+	bt_mesh_crypto_init();
+	pp_netkey_ctx_init(&pp_net0);
+	k_sem_init(&observer_sem, 0, 1);
+
+	err = bt_enable(NULL);
+	if (err) {
+		FAIL("Bluetooth init failed (err %d)", err);
+		return;
+	}
+}
+
+static void tx_priv_common_init(uint16_t wait)
+{
+	int err;
+
+	bt_mesh_test_cfg_set(NULL, wait);
+	bt_mesh_device_setup(&prov, &prb_comp);
+	provision(&tx_cfg);
+
+	/* Disable GATT proxy */
+	err = bt_mesh_gatt_proxy_set(BT_MESH_GATT_PROXY_DISABLED);
+	if (err) {
+		FAIL("Failed to disable gatt proxy (err %d)", err);
+	}
+}
+
+static void test_tx_priv_net_id(void)
+{
+	int err;
+
+	tx_priv_common_init(610);
+
+	/* Enable private GATT proxy */
+	err = bt_mesh_priv_gatt_proxy_set(BT_MESH_GATT_PROXY_ENABLED);
+	if (err) {
+		FAIL("Failed to set private gatt proxy (err %d)", err);
+	}
+
+	PASS();
+}
+
+static void test_rx_priv_net_id(void)
+{
+	struct priv_test_ctx ctx = {
+		.beacon_type = BEACON_TYPE_PRIVATE_NET_ID,
+		.node_id_addr = NULL,
+		.arg.same_random = false,
+	};
+
+	rx_priv_common_init(610);
+
+	/* Scan for first net ID */
+	ASSERT_TRUE(
+		wait_for_beacon(priv_scan_cb, 1, pp_beacon_check, &ctx));
+
+	/* Wait for 10 minutes, then scan for another net
+	 * ID beacon and verify that random field has changed
+	 */
+	k_sleep(K_SECONDS(600));
+	ASSERT_TRUE(
+		wait_for_beacon(priv_scan_cb, 1, pp_beacon_check, &ctx));
```

Comment:
Can the random field be checked here rather than in `pp_beacon_check`? Currently, if the test fails, the error message will say that the test didn't receive the beacon. If we compare the random fields here, the error will say that random fields are the same (or not) which is much more clear.